### ISSUE TITLE
Null-safe version not recognized as such

### ISSUE THREAD

**Issue by amake** [petitparser/dart-petitparser#89]
I started looking into migrating a package of mine that depends on PetitParser, and I found that the tooling does not recognize 4.0.0-nullsafety as being null-safe:

```sh
% dart pub outdated --mode=null-safety
Showing dependencies that are currently not opted in to null-safety.
[✗] indicates versions without null safety support.
[✓] indicates versions opting in to null safety.

Package Name  Current  Upgradable  Resolvable             Latest                 

direct dependencies:
petitparser   ✗2.4.0   ✗2.4.0      ✗3.1.0                 ✗4.0.0-nullsafety      

dev_dependencies:
test          ✗1.15.6  ✗1.15.6     ✓1.16.0-nullsafety.11  ✓1.16.0-nullsafety.11 
```

It looks like this is due to your `sdk` version spec:

https://github.com/petitparser/dart-petitparser/blob/f70568bc3fa52301ca68a4e0c912bbe3b692d993/petitparser/pubspec.yaml#L8-L9

The [migration guide says](https://dart.dev/null-safety/migration-guide#migrating-by-hand) you need

```yaml
environment:
  sdk: '>=2.12.0-0 <3.0.0'
```

**Comment by renggli** [comment]
The currently published null-safe version works with the stable version of Dart. I can try to publish an updated version later this weekend, but likely it just works if you ignore the warning.

Another problem is that some dev_dependencies do not have published null-safe versions. Again, it works if you ignore such warnings.

**Comment by renggli** [comment]
I've published [4.0.0-nullsafety.1](https://pub.dev/packages/petitparser/versions/4.0.0-nullsafety.1) which should have the recommended `sdk` spec for Dart null safety beta.
